demo/cmd/common/books.go

```diff
@@ -29,6 +31,18 @@ const (
 	httpsPrefix = "https://"
 )
 
+// GetServiceURLBasedOnWorkerOS  is used to retried the service url based on the worker OS.
+func GetServiceURLBasedOnWorkerOS(namespace string, serviceName string, port int) string {
+	postfix := ""
+	if runtime.GOOS == constants.OSWindows {
+		// Due to a limitation on kubernetes on Windows we need to add the postfix
+		// otherwise DNS will not be able to resolve it.
+		// https://kubernetes.io/docs/setup/production-environment/windows/intro-windows-in-kubernetes/#dns-limitations
+		postfix = ".svc.cluster.local"
```

Comment:
I don't see any issue with adding this unconditionally if we can get rid of a `if windows` branch.

Reply:
I remember that @shashankram had some concerns when I mentioned this in a standup a month back this is why I made it conditional.

No strong opinion.

Reply:
Hey @davinci26 my concern was regarding using windows defaults for linux in e2e tests, which could introduce testing gaps in linux. But in this instance, it makes sense to remove this conditional and use the same URL for windows and linux, with a note on why an FQDN must be used.